### PR TITLE
[1812] initial commit

### PR DESCRIPTION
**Pull request by philcampeau**
## Before clicking "Create"

- [x] Branch is derived from the latest `master`
- [x] Add the `pins` or `archive_alpha_games` label if this change will break existing games
- [x] Code passes linter with `docker compose exec rack rubocop -a`
- [x] Tests pass cleanly with `docker compose exec rack rake`

### Explanation of Change

Initial map, meta, entities, and game file, plus a couple of steps. Entities are functional.

### CODE REVIEW

**Comment by ollybh** [comment]
@philcampeau Can you please hold off merging your pull requests until others have had a change to review them?
1812 is an 1861 derivative, you're likely to find this easier if G1812 is a subclass of G1867.

**Comment by philcampeau** [comment]
Sure, not a problem. I figured this was basically just the shell of the game so I didn't really need comments on it. I was planning to use 1861 as a base when I really start working on it.

File: lib/engine/game/g_1812/entities.rb [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+                         # because this is only for specific trains, it should be coded in def revenue_for, I think.
+                       ],
+          },
+          {
+            name: 'Sheffeild & Rotherham Railway',
+            sym: 'P4',
+            value: 20,
+            min_price: 15,
+            revenue: 5,
+            desc: 'A corporation owning the Sheffeild & Rotherham Railway gains a +£10 bonus for all runs to Sheffield (C18).',
```
Comment (ollybh) [inline lib/engine/game/g_1812/entities.rb:48]: Sheffield, not Sheffeild.

File: lib/engine/game/g_1812/entities.rb [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+                          cost: 0,
+                        }],
+          },
+          {
+            name: 'Hull Docks Company',
+            sym: 'P11',
+            value: 40,
+            min_price: 30,
+            revenue: 10,
+            desc: 'A corporation owning the Newcastle & Carlisle Railway gains a +£20 bonus for all runs to Hull (G16).',
```
Comment (ollybh) [inline lib/engine/game/g_1812/entities.rb:193]: Wrong private name in the description.

File: lib/engine/game/g_1812/entities.rb [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+                          when: 'never',
+                        }],
+          },
+          {
+            name: 'Hartlepool Docks Company',
+            sym: 'P12',
+            value: 40,
+            min_price: 30,
+            revenue: 10,
+            desc: 'A corporation owning the Hull Docks Company gains a +£20 bonus for all G-train runs to Hartlepool '\
```
Comment (ollybh) [inline lib/engine/game/g_1812/entities.rb:211]: Wrong private name in the description.

File: lib/engine/game/g_1812/entities.rb [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+            tokens: [0],
+            shares: [100],
+            type: 'minor',
+            coordinates: 'C14',
+            city: 0,
+            color: '#777777',
+          },
+          {
+            sym: '7',
+            name: '	North Midland Railway',
```
Comment (ollybh) [inline lib/engine/game/g_1812/entities.rb:299]: There's a stray space in the company name.

**Comment by ianwilson156** [comment]
If you need help with checking or testing, just ask! I don't code but I've been told I'm good at pre-alpha testing to find bugs (in hotseat mode, usually).

**Pull request merged**